src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -32,6 +33,91 @@ internal static partial class PathInternal
         };
 
         /// <summary>
+        /// Adds the extended path prefix (\\?\) if not already present.
+        /// </summary>
+        internal static string AddExtendedPathPrefix(string path)
+        {
+            if (IsExtended(path))
+                return path;
+
+            if (IsExtendedUnc(path))
+                return path.Insert(2, PathInternal.UncExtendedPrefixToInsert); // Given \\server\share in longpath becomes \\?\UNC\server\share  => UNCLongPathPrefix + path.SubString(2); => The actual command simply reduces the operation cost.
```

Comment:
But I'm confused... aren't we at this return statement if we've already verified this is already an extended unc path?  Why are we then inserting the extended prefix again?